istioctl/cmd/describe.go

```diff
@@ -427,6 +428,14 @@ func printPod(writer io.Writer, pod *v1.Pod) {
 			}
 			ports = append(ports, fmt.Sprintf("%d%s (%s)", port.ContainerPort, protocol, container.Name))
 		}
+		// Ref: https://istio.io/latest/docs/ops/deployment/requirements/#pod-requirements
+		if container.Name != "istio-proxy" {
```

Comment:
Why is `describe` the right place for this? I would think a warning in the webhook, `analyze`, a log in istiod, etc are all better places?

Reply:
> Why is describe the right place for this?

I think users are expecting to show a warning related to pod config in `describe` as mentioned in this issue https://github.com/istio/istio/issues/31874.

> I would think a warning in the webhook, analyze, a log-in istiod, etc are all better places?

I agree we should also add the warning to `analyze` and istiod logs.

Reply:
@esnible Added Analyzer for the same https://github.com/istio/istio/pull/31993